unidist/core/backends/mpi/core/worker/task_store.py

```diff
@@ -249,22 +250,31 @@ async def execute():
                             isinstance(output_data_ids, (list, tuple))
                             and len(output_data_ids) > 1
                         ):
+                            completed_data_ids = output_data_ids
```

Comment:
We should update completed_data_ids in the loop below after getting a unique data id, but not here.